spec.html

```diff
@@ -24010,6 +24172,154 @@ <h1>Static Semantics: ContainsArguments</h1>
       </emu-alg>
     </emu-clause>
 
+    <emu-clause id="sec-static-semantics-containsawait" type="sdo" aoid="ContainsAwait">
+      <h1>Static Semantics: ContainsAwait</h1>
```

Comment:
Aside: it is unfortunate we don't have a predicate-taking version of Contains. If this pattern comes up a few more times maybe we can refactor.

Reply:
I should say there are quite a few of these ContainsFoo SDOs already, so the "comes up a few more times" threshold might have already been met? And maybe it wouldn't save that much space or readability anyway.

Reply:
In the TLA PR we [decided](https://github.com/tc39/ecma262/pull/2408#discussion_r671918604) we were OK with using ``Contains `await`.`` I still think that's a bit suspicious, but since it's precedented now, I suppose we can use it here instead of the new SDO.

Reply:
Ah true, I had forgotten that.

Reply:
As long as this catches `for await` (which I think it does, given the definition of Contains), then I'm happy to remove this.